clar2wasm/src/error_mapping.rs

```diff
@@ -16,7 +16,7 @@ pub enum ErrorMap {
     DivisionByZero = 2,
     ArithmeticLog2Error = 3,
     ArithmeticSqrtiError = 4,
-    UnwrapFailure = 5,
+    BadTypeConstruction = 5,
     Panic = 6,
     ShortReturnAssertionFailure = 7,
     ArithmeticPowError = 8,
```

Comment:
Any chance you could add some documentation for all the enum variants? No idea what `BadTypeConstruction` means from afar.

Reply:
Sure thing. Done.

Reply:
Nice documentation!